### PR TITLE
Added shim for Spark 3.3.2

### LINKED ISSUE #6902: [FEA] Add shims for Spark 3.3.2
The latest version of Spark 3.3 branch [is "3.3.2-SNAPSHOT"](https://github.com/apache/spark/blob/branch-3.3/pom.xml#L29).

**Additional context**
https://github.com/NVIDIA/spark-rapids/issues/6896

### PR DESCRIPTION
**Pull request by razajafri**
Added shim for Spark-3.3.2 as this is the new version for Spark 3.3 

fixes #6902 

Signed-off-by: Raza Jafri <rjafri@nvidia.com>

### CODE REVIEW

**Review by jlowe — COMMENTED** [review]
`snapshot.buildvers` needs to be updated in dist/pom.xml

**Review by jlowe — DISMISSED** [review]

**Comment by jlowe** [comment]
build

**Comment by pxLi** [comment]
either this one or https://github.com/NVIDIA/spark-rapids/pull/6920 would resolve merge conflict after merging another one

**Comment by razajafri** [comment]
build

File: pom.xml [hunk 1]
```diff
@@ -1105,6 +1173,7 @@
         <spark321db.version>3.2.1-databricks</spark321db.version>
         <spark322.version>3.2.2</spark322.version>
         <spark330.version>3.3.0</spark330.version>
+        <spark332.version>3.3.2-SNAPSHOT</spark332.version>
         <spark331.version>3.3.1</spark331.version>
```
Comment (jlowe) [inline pom.xml:1177]: Nit: Please keep these ordered.
```suggestion
        <spark331.version>3.3.1</spark331.version>
        <spark332.version>3.3.2-SNAPSHOT</spark332.version>
```

**Comment by razajafri** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**